qiskit/visualization/timeline/drawing_objects.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""
+Drawing objects for the timeline drawer.
+
+Drawing objects play two important roles:
+    - Enabling unittests of the visualization module.
+    - Removing program parser from each plotter interface. We can now easily add new plotters.
+
+This module is based on the structure of matplotlib as it is the primary plotter
+of the timeline drawer. However this interface is agnostic to the actual plotter implementation.
+
+Design concept
+~~~~~~~~~~~~~~
+When we think about dynamically updating drawing objects, it will be most efficient to
+update only the changed properties of drawings rather than regenerating entirely from scratch.
+Thus the core drawing function generates all possible drawings in the beginning and
+then updates the visibility and the offset coordinate of each item according to
+the end-user request.
+
+Data key
+~~~~~~~~
+In the abstract class ``ElementaryData`` common properties that represent a drawing object are
+specified. In addition, drawing objects have the `data_key` property that returns a
+unique hash of the object for comparison. This property should be defined in each sub-class by
+considering necessary properties to identify that object, i.e. `visible` should not
+be a part of the key, because any change on this property just sets the visibility of
+the same drawing object.
+
+To support other plotters than matplotlib, drawing objects should be
+universal and designed without strong dependency on modules in `matplotlib`.
+This means drawing objects that represent primitive geometries are preferred.
+It should be noted that there will be no unittest for a plotter interface, which takes
+drawing objects and output an image data, we should avoid adding a complicated data structure
+that has a context of the scheduled circuit program.
+
+Usually a drawing object is associated to the specific bit. `BitLinkData` is the
+exception of this framework because it takes multiple bits to connect.
+Position of the link is dynamically updated according to the user preference of bit to show.
+While other objects are general format to represent specific shapes, the `BitLinkData` is
+only used by the bit links and thus has no `data_type` input.
```

Comment:
This text already exists in `qiskit/visualization/pulse_v2/drawing_objects.py` Could you reference it instead of repeating it?

Reply:
I updated a text. Eventually I want to make some base plotter module to provide a skeleton and write everything there.